src/AutoRest.CSharp/Common/Input/InputTypes/Transformation/InputRenameTimeToOnTransformer.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using AutoRest.CSharp.Output.Builders;
+
+namespace AutoRest.CSharp.Common.Input
+{
+    internal class InputRenameTimeToOnTransformer
+    {
+        private static readonly Dictionary<string, string> _nounToVerbDicts = new()
+        {
+            {"Creation", "Created"},
+            {"Deletion", "Deleted"},
+            {"Expiration", "Expire"},
+            {"Modification", "Modified"},
+        };
+
+        public static void Update(InputNamespace inputNamespace)
+        {
+            foreach (var model in inputNamespace.Models)
+            {
+                if (model is not InputModelType inputModel)
+                    continue;
+
+                foreach (var property in inputModel.Properties)
+                {
+                    var propertyType = property.Type.GetImplementType();
+                    if ((propertyType is InputPrimitiveType inputPrimitiveType && inputPrimitiveType.Kind == InputPrimitiveTypeKind.PlainDate) || propertyType is InputDateTimeType)
+                    {
+                        var propName = property.CSharpName();
+
+                        if (propName.StartsWith("From", StringComparison.Ordinal) ||
+                            propName.StartsWith("To", StringComparison.Ordinal) ||
+                            propName.EndsWith("PointInTime", StringComparison.Ordinal))
+                            continue;
+
+                        var lengthToCut = 0;
+                        if (propName.Length > 8 &&
+                            propName.EndsWith("DateTime", StringComparison.Ordinal))
+                        {
+                            lengthToCut = 8;
+                        }
+                        else if (propName.Length > 4 &&
+                            propName.EndsWith("Time", StringComparison.Ordinal) ||
+                            propName.EndsWith("Date", StringComparison.Ordinal))
+                        {
+                            lengthToCut = 4;
+                        }
+                        else if (propName.Length > 2 &&
+                            propName.EndsWith("At", StringComparison.Ordinal))
+                        {
+                            lengthToCut = 2;
+                        }
+
+                        if (lengthToCut > 0)
+                        {
+                            var prefix = propName.Substring(0, propName.Length - lengthToCut);
+                            var newName = (_nounToVerbDicts.TryGetValue(prefix, out var verb) ? verb : prefix) + "On";
+                            property.Name = newName;
+                        }
```

Comment:
since we are previously doing this in this way, this is not a comment that we have to change it right now.
But we have issues on the current implementation.
For example, we are checking `StartsWith("To", `, what if there was a word starting with `To` but it is actually not the word `To`? To be precise, this should be implemented in a way of "words" instead of just checking if the whole string starts with what or ends with what.

We might need to fix this when we migrate this logic to MGC.

Reply:
Also we could also consider to change the implementation of these to using spans, which should improves the performance.

Reply:
updated